### PR TITLE
Allow submitting other educator role name

### PR DESCRIPTION
**Pull request by BryanEliDimas**
QA [found a problem](https://openstax.slack.com/archives/C6NRD0CCQ/p1590175423228100) with submitting Educator step 4 form. This PR fixes the immediate problem.

The long term problem will be solved by https://github.com/openstax/business-intel/issues/1144

### CODE REVIEW

File: app/controllers/newflow/login_signup_controller.rb [hunk 1]
```diff
@@ -80,6 +80,9 @@ def educator_complete
           redirect_to signup_done_path
         },
         failure: lambda {
+          @book_subjects = book_data.subjects
```
Comment (steveburkett) [inline app/controllers/newflow/login_signup_controller.rb:83]: is book_data guaranteed to not be nil?

File: app/handlers/newflow/educator_complete_profile.rb [hunk 2]
```diff
@@ -67,6 +60,10 @@ def push_lead_to_salesforce
       end
     end
 
+    def kind_of_educator
+      signup_params.educator_specific_role == 'other' ? signup_params.other_role_name : signup_params.educator_specific_role
```
Comment (steveburkett) [inline app/handlers/newflow/educator_complete_profile.rb:64]: consider putting 'other' as a constant.

consider a method name of `educator_type`? (subtle..just wondering if this sounds better than "kind_of". either way is fine :)

Comment (steveburkett) [reply app/handlers/newflow/educator_complete_profile.rb:64]: does the role need to be stripped? (any chance this role could come in w/ leading or trailing spaces?  it's good to nip this in the bud if so and trim the inputs (always)

File: app/controllers/newflow/login_signup_controller.rb (hunk 1 above)
Comment (BryanEliDimas) [reply app/controllers/newflow/login_signup_controller.rb:83]: Not really but that is something I am working on improving in a separate PR: https://github.com/openstax/accounts/pull/840

File: app/handlers/newflow/educator_complete_profile.rb (hunk 2 above)
Comment (BryanEliDimas) [reply app/handlers/newflow/educator_complete_profile.rb:64]: These are good ideas I would have implemented before hitting the merge button. I can definitely implement them in another PR which I'm currently working on unless you think I should reverse this PR

**Pull request merged**